### PR TITLE
Refractor to serve JS locally

### PR DESCRIPTION
**Pull request by theskumar**
Fixes: None

Content served CDN is an area of concern for privacy & performance.
Ensure all the JS code is served locally.

- Serve ‘url-search-params.js’ locally.
- Serve ‘symbol-es6.min.js’ locally.
- Serve “jquery.fancybox.min.js" locally.

### CODE REVIEW

**Comment by theskumar** [comment]
I need to look at the lint issues. Do not merge.

File: hypha/static_src/src/javascript/apply/url-search-params.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
...
...(value){items.push(value)});var iterator={next:function(){var value=items.shift();return{done:value===undefined,value:value}}};if(iterable){iterator[Symbol.iterator]=function(){return iterator}}return iterator}}if(!("entries"in URLSearchParamsProto)){URLSearchParamsProto.entries=function entries(){var items=[];this.forEach(function(value,name){items.push([name,value])});var iterator={next:function(){var value=items.shift();return{done:value===undefined,value:value}}};if(iterable){iterator[Symbol.iterator]=function(){return iterator}}return iterator}}if(iterable&&!(Symbol.iterator in URLSearchParamsProto)){URLSearchParamsProto[Symbol.iterator]=URLSearchParamsProto.entries}if(!("sort"in URLSearchParamsProto)){URLSearchParamsProto.sort=function sort(){var entries=this.entries(),entry=entries.next(),done=entry.done,keys=[],values=Object.create(null),i,key,value;while(!done){value=entry.value;key=value[0];keys.push(key);if(!(key in values)){values[key]=[]}values[key].push(value[1]);entry=entries.next();done=entry.done}keys.sort();for(i=0;i<keys.length;i++){this["delete"](keys[i])}for(i=0;i<keys.length;i++){key=keys[i];this.append(key,values[key].shift())}}}})(URLSearchParams.prototype);
```
[the remark this answers is not in the document]
Comment (theskumar) [reply hypha/static_src/src/javascript/apply/url-search-params.js]: Patching it up with the latest version.

File: hypha/apply/projects/templates/application_projects/overview.html [hunk 2]
```diff
@@ -75,11 +75,11 @@ <h5>{% trans "Track and explore recent projects" %}</h5>
 
 {% block extra_js %}
     {{ projects.filterset.form.media.js }}
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.4.1/jquery.fancybox.min.js"></script>
+    <script src="https:{% static 'js/apply/jquery.fancybox.min.js' %}"></script>
```
Comment (frjo) [inline hypha/apply/projects/templates/application_projects/overview.html]: Quick fix here to remove "https:".

**Review by frjo — CHANGES_REQUESTED** [review]
Works well in my testing, just one small issue.

**Comment by frjo** [comment]
Easy way to get the select2 js and css to also be local?

File: hypha/apply/projects/templates/application_projects/overview.html (hunk 2 above)
Comment (theskumar) [reply hypha/apply/projects/templates/application_projects/overview.html]: Good catch, will fix it.

Comment (theskumar) [inline hypha/apply/projects/templates/application_projects/overview.html]:
```suggestion
    <script src="{% static 'js/apply/jquery.fancybox.min.js' %}"></script>
```

**Comment by theskumar** [comment]
> Easy way to get the select2 js and css to also be local?

Are you talking about the `django-select2` package? They do have css/js in their `static` local folder. Would you be able to point me where it's you see them remotely served?

**Comment by theskumar** [comment]
Other than select2 issue I believe this is good to merge in, I can take a stab at select2 js/css in another PR.

**Review by frjo — APPROVED** [review]

**Pull request merged**